### DISCUSSION TITLE
Undefined variable $form in edit view

### DISCUSSION THREAD

**Discussion by kokoshneta** [filamentphp/filament#4248] [Help] [answered]
I started implementing Filament in a project a while ago and then left it alone for a while, having only implemented a few models so far. The implementation for those models has been working, and in an earlier version of the project deployed to the server, it works just fine.

But on my localhost, where I’ve continued developing, the edit pages for all the models I’ve implemented now throws the error »Undefined variable $form« on line 25 in `edit-record.blade.php`. This is about ten lines after `$form` is defined using a Blade [@capture directive](https://github.com/ryangjchandler/blade-capture-directive):

```
    @capture($form) // <-- this should define $form
        <x-filament::form wire:submit.prevent="save">
            {{ $this->form }}

            <x-filament::form.actions
                :actions="$this->getCachedFormActions()"
                :full-width="$this->hasFullWidthFormActions()"
            />
        </x-filament::form>
    @endcapture

    @php
        $relationManagers = $this->getRelationManagers();
    @endphp

    @if ((! $this->hasCombinedRelationManagerTabsWithForm()) || (! count($relationManagers)))
        {{ $form() }} // <-- this line throws the error
    @endif
```

I have modified various things in some of the models the edit views draw on, but as far as I can tell, nothing that is used in the `[Model]Resource::form()` function. The list view (based on `[Model]Resource::table()` works just fine.

Googling the error yields pretty much nothing at all, and since I’m only just getting started at implementing Filament, I’m still a long way from actually understanding it well enough to even guess at how this can be.

What might be causing `$form` to not get set with the `@capture` directive?

**Comment by danharrin** [comment, marked as answer]
Have you published the views or anything? What Laravel version are you using?

**Reply by kokoshneta** [reply]
I don’t believe I have published the views (though I’m not entirely sure). At least I’m fairly sure that if I did ever publish the views, I did so _before_ the earlier version was deployed to the server. The view file is located at `/vendor/filament/filament/resources/views/resources/pages/edit.record.blade.php`.

I did publish some other files, but I think only the translations.

I’m on Laravel 9.31.0 running on a MacBook Pro using Valet.

**Reply by danharrin** [reply]
And if you create a new project, install Filament, can you reproduce this error in your environment? Have you run `php artisan filament:upgrade`? Maybe delete composer.lock and /vendor and run composer update again?

**Reply by kokoshneta** [reply]
Huh, whaddya know… `filament:upgrade` did the trick. I’d done `optimize:clear`, but I guess that doesn’t take care of Filament’s caches. So I guess it was just a cache issue somehow. Thanks!

**Comment by pablorsk** [comment]
Just in case...

If you dont have package autodiscover, check if you have `\RyanChandler\BladeCaptureDirective\BladeCaptureDirectiveServiceProvider::class` added on `config/app.php`.
